lang/rust/avro/tests/shared.rs

```diff
@@ -0,0 +1,84 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use apache_avro::{types::Value, Codec, Reader, Schema, Writer};
+use std::{
+    fs::{DirEntry, File, ReadDir},
+    io::{BufReader, Error},
+    path::Path,
+    slice::Iter,
+};
+
+const ROOT_DIRECTORY: &str = "../../../share/test/data/schemas";
+
+#[test]
+fn test_schema() {
+    let directory: ReadDir = scan_shared_folder();
+    directory.for_each(|f: Result<DirEntry, Error>| {
+        let e: DirEntry = match f {
+            Err(error) => panic!("Can't get file {:?}", error),
+            Ok(entry) => entry,
+        };
+        log::warn!("{:?}", e.file_name());
+        let sub_folder = ROOT_DIRECTORY.to_owned() + "/" + e.file_name().to_str().unwrap();
+
+        test_folder(sub_folder.as_str());
+    });
+}
+
+fn test_folder(folder: &str) {
+    let file_name = folder.to_owned() + "/schema.json";
+    let content = std::fs::read_to_string(file_name).expect("Unable to find schema.jon file");
```

Comment:
I think it would be better if `test_folder()` returns a `Result<(), Error>`. This way all `expect()` calls could be simplified to `?` and we could see the real error when it happens.